utils/cache.js

```diff
@@ -0,0 +1,22 @@
+(function (root, factory) {
+    if (typeof module === 'object' && module.exports) {
+        module.exports = factory(require('./string'));
+    } else {
+        root['autoComplete/utils/cache'] = factory(root['autoComplete/utils/string']);
+    }
+}(typeof self !== 'undefined' ? self : this, function (_string) {
+    var clearString = _string.clearString;
+
+    function removeSuggestionFromCache(cache, suggestion) {
+        var clean_cache = {};
```

Comment:
`snake_case` is not common in JS, unless it is all in uppercase for constants.
```suggestion
        var cleanCache = {};
```